# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/PersistentBufferProvider.cpp
# Commit: 74e70f651dca
# Full Hash: 74e70f651dcacc39e1c42524a6afdeb5f51fb717
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-01-06 04:14:44
# Description:
#   Bug 1808733 - Check for null TextureData. r=aosmond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D166093
# ==============================================================================

diff -r 48ca857f942d -r 74e70f651dca gfx/layers/PersistentBufferProvider.cpp
--- a/gfx/layers/PersistentBufferProvider.cpp	Thu Jan 05 17:37:58 2023 +0000
+++ b/gfx/layers/PersistentBufferProvider.cpp	Thu Jan 05 17:45:48 2023 +0000
@@ -656,10 +656,11 @@
 bool PersistentBufferProviderShared::IsAccelerated() const {
 #ifdef XP_WIN
   // Detect if we're using D2D canvas.
-  if (mWillReadFrequently || mTextures.empty()) {
+  if (mWillReadFrequently || mTextures.empty() || !mTextures[0]) {
     return false;
   }
-  if (mTextures[0]->GetInternalData()->AsD3D11TextureData()) {
+  auto* data = mTextures[0]->GetInternalData();
+  if (data && data->AsD3D11TextureData()) {
     return true;
   }
 #endif
